lib/web/app/transport.go

```diff
@@ -159,39 +165,62 @@ func (t *transport) rewriteRequest(r *http.Request) error {
 	return nil
 }
 
-// websocketsDialer returns a function that dials a websocket connection
-// over the transport's reverse tunnel.
-func websocketsDialer(tr *http.Transport) forward.Dialer {
-	return func(network, address string) (net.Conn, error) {
-		conn, err := tr.DialContext(context.Background(), network, address)
+// DialContext
+func (t *transport) DialContext(ctx context.Context, _ string, _ string) (net.Conn, error) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+
+	for i := len(t.c.servers) - 1; i >= 0; i-- {
+		conn, err := dialAppServer(t.c.proxyClient, t.c.identity, t.c.servers[i])
 		if err != nil {
+			// Connection problem with the server.
+			if trace.IsConnectionProblem(err) {
+				t.c.log.Warnf("failed to connect to app server: %v.", err)
```

Comment:
```suggestion
				t.c.log.Warnf("Failed to connect to app server: %v.", err)
```

Also, add server ID to the message.